052-cruise-control-api-users.md

```diff
@@ -0,0 +1,261 @@
+
+Strimzi would decode and use the contents of this secret to populate the Cruise Control API auth credentials file which would be mounted and used by the Cruise Control pod.
+
+Although this would simplify the overall design it would also shift a lot of the maintanence responsibility to the developer.
+The developer would need to create a valid Cruise Control API user configuration and encode it into a secret.
+That being said, this is still a reasonable amount of responsibility to leave to the developer.
+This is because accessing the Cruise Control API directly is neither the supported or primary way of interacting with Cruise Control for a Strimzi-managed Kafka cluster.
+
+### Risks
+
+Of course, supporting such a feature could come with some risks.
+
+If we were to support developers to create Cruise Control API users with the `ADMIN` role, those users would have access to potentially destructive cluster operations.
+For example, an API user with the ADMIN role would be able to execute a rebalance opeation while the Strimzi Operator is in the middle of preforming a rolling update of the Kafka brokers.
+
+For the above reasons, this proposal advocates for disabling the creation of API users with `ADMIN` roles for now.
+
+We can always enable the `ADMIN` role at a later date with a caveat in the documentation so that advanced developers could experiment with Cruise Control write operations at their own risk.
+
+That being said, supporting developers to create Cruise Control API users with non-`ADMIN` roles like `USER` and `VIEWER` roles would mitigate this risk completely.
+This is because the `USER` and `VIEWER` roles are read-only roles and would not conflict with Strimzi cluster operations.
+So this proposal if for enabling the `USER` and `VIEWER` roles only.
+
+### How API user management is handled elsewhere
+
+Cruise Control is not the only Kafka component which has API users.
+For consitency across the code base, we should also consider the following components when designing our solution:
+
+* Kafka users
+* KafkaConnect users
+
+#### Kafka users
+
+In Strimzi, Kafka users are managed via custom resources.
+
+```yaml=
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaUser
+metadata:
+  name: my-user-1
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  authentication:
+    type: tls
+  authorization:
+    type: simple
+    acls:
+      ...
+```
+These resources are watched and managed by the the Strimzi User Operator, which creates Kafka users based on the descriptions provided by the resources.
+
+For more details, checkout the Strimzi documentation [here](https://strimzi.io/docs/operators/latest/deploying#proc-configuring-kafka-user-str)
+
+#### KafkaConnect users
+
+At this time, Strimzi doesn't offer any method of creating or configuring KafkaConnect API users.
+That being said, the direction we take with creating and managing Cruise Control API users could be used as a precedent for its design in the future.
```

Comment:
The flow of this seems a bit weird as you already covered how it should be handled in CC. But I guess it might be useful for some readers? Should we move it to the _Current Situation_ section? Or should we frame it more how does the CC user management compare with other parts of Strimzi?

Reply:
Yeah, I might be tempted to move it between Current Situation and Motivation. But say that the proposal is only about CC for now in the moved section.